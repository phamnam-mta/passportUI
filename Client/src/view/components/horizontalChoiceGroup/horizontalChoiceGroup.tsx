import * as React from "react";
import { ChoiceGroup, IChoiceGroupProps } from "@fluentui/react";

export interface IHorizontalChoiceGroupProps extends IChoiceGroupProps {}

export class HorizontalChoiceGroup extends React.PureComponent<IHorizontalChoiceGroupProps> {
    public render() {
        return (
            <ChoiceGroup
                {...this.props}
                styles={{
                    root: { selectors: { "& > div": { display: "flex", alignItems: "center", gap: 12 } } },
                    flexContainer: {
                        display: "flex",
                        gap: 12,
                        selectors: {
                            "& .ms-ChoiceField": {
                                marginTop: 0,
                            },
                        },
                    },
                }}
            />
        );
    }
}
