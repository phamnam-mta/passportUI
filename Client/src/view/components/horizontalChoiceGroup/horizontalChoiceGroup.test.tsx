import * as React from "react";
import { shallow } from "enzyme";

import { HorizontalChoiceGroup } from "./horizontalChoiceGroup";

describe("<HorizontalChoiceGroup />", () => {
    let baseProps;
    beforeEach(() => {});

    describe("Rendering", () => {
        it("renders correctly", () => {
            const wrapper = shallow(<HorizontalChoiceGroup {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
