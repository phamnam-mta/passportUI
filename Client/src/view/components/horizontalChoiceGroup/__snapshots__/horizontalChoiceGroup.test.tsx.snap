// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorizontalChoiceGroup /> Rendering renders correctly 1`] = `
<StyledChoiceGroup
  styles={
    Object {
      "flexContainer": Object {
        "display": "flex",
        "gap": 12,
        "selectors": Object {
          "& .ms-ChoiceField": Object {
            "marginTop": 0,
          },
        },
      },
      "root": Object {
        "selectors": Object {
          "& > div": Object {
            "alignItems": "center",
            "display": "flex",
            "gap": 12,
          },
        },
      },
    }
  }
/>
`;
